Libraries/MiscDrivers/ExtMemory/is25.c

```diff
@@ -0,0 +1,616 @@
+        return err;
+    }
+
+    if ((err = Ext_Flash_Read_SR(&buf, Ext_Flash_StatusReg_1)) != EF_E_SUCCESS) {
+        return err;
+    }
+
+    if (buf & IS25_WEL_MASK) {
+        return EF_E_SUCCESS;
+    }
+
+    return EF_E_BAD_STATE;
+}
+
+/* ************************************************************************* */
+#if defined(SPIXF_RAM) && IAR_PRAGMAS
+#pragma section = ".spix_config"
+#elif defined(SPIXF_RAM)
+__attribute__((section(".spix_config")))
+#endif
+static int inline read_reg(uint8_t cmd, uint8_t *buf)
+{
+    int err = EF_E_SUCCESS;
+
+    if (!buf) {
+        return EF_E_BAD_PARAM;
+    }
+
+    // Send the command
+    if ((err = g_cfg.write(&cmd, 1, 0, Ext_Flash_DataLine_Single)) != EF_E_SUCCESS) {
+        return err;
+    }
+
+    // Read the data
+    if ((err = g_cfg.read(buf, 1, 1, Ext_Flash_DataLine_Single)) != EF_E_SUCCESS) {
+        return err;
+    }
+
+    return EF_E_SUCCESS;
+}
+
+/* ************************************************************************* */
+#if defined(SPIXF_RAM) && IAR_PRAGMAS
+#pragma section = ".spix_config"
+#elif defined(SPIXF_RAM)
+__attribute__((section(".spix_config")))
+#endif
+static int inline write_reg(uint8_t *buf, unsigned len)
+{
+    int err = EF_E_SUCCESS;
+
+    if (!buf || (len == 0)) {
+        return EF_E_BAD_PARAM;
+    }
+
+    if ((err = write_enable()) != EF_E_SUCCESS) {
+        return err;
+    }
+
+    // Send the command and data
+    if ((err = g_cfg.write(buf, len, 1, Ext_Flash_DataLine_Single)) != EF_E_SUCCESS) {
+        return err;
+    }
+
+    return EF_E_SUCCESS;
+}
+
+/* **** Functions **** */
+
+int Ext_Flash_Configure(Ext_Flash_Config_t *cfg)
+{
+    int err = EF_E_SUCCESS;
+
+    if (cfg == NULL) {
+        return EF_E_BAD_PARAM;
+    }
+
+    g_cfg = *cfg;
+    g_is_configured = 1;
+
+    return err;
+}
+
+/* ************************************************************************* */
+#if defined(SPIXF_RAM) && IAR_PRAGMAS
+#pragma section = ".spix_config"
+#elif defined(SPIXF_RAM)
+__attribute__((section(".spix_config")))
+#endif
+int Ext_Flash_Init(void)
+{
+    if (!g_is_configured) {
+        return EF_E_BAD_STATE;
+    }
+
+    return g_cfg.init();
+}
+
+/* ************************************************************************* */
+#if defined(SPIXF_RAM) && IAR_PRAGMAS
+#pragma section = ".spix_config"
+#elif defined(SPIXF_RAM)
+__attribute__((section(".spix_config")))
+#endif
+int Ext_Flash_Reset(void)
+{
+    int err = EF_E_SUCCESS;
+    int busy_count = 0;
+    uint8_t cmd = IS25_CMD_RST_EN;
+
+    // Send the Reset command
+    if ((err = g_cfg.write(&cmd, 1, 1, Ext_Flash_DataLine_Single)) != EF_E_SUCCESS) {
+        return err;
+    }
+
+    cmd = IS25_CMD_RST_MEM;
+    if ((err = g_cfg.write(&cmd, 1, 1, Ext_Flash_DataLine_Single)) != EF_E_SUCCESS) {
+        return err;
+    }
+
+    while (flash_busy()) {
+        busy_count++;
+        if (busy_count > 20000) {
+            return EF_E_TIME_OUT;
```

Comment:
if (++busy_count > 2000)
    return ER...

Reply:
Existing one is less complex to read. Compiler will generate same binary.

Reply:
No changes made.